source/_integrations/flexmeasures.markdown

```diff
@@ -0,0 +1,156 @@
+---
+title: FlexMeasures
+description: Use FlexMeasures flexible energy asset in Home Assistant 
+ha_category:
+  - Energy
+ha_release: "2023.9"
+ha_iot_class: Cloud Polling
+ha_codeowners:
+  - '@victorgarcia98'
+  - '@GustaafL'
+  - '@Flix6x'
+ha_domain: flexmeasures
+ha_platforms:
+  - sensor
+ha_integration_type: integration
+ha_config_flow: true
+ha_quality_scale: silver
+---
+
+
+
+
+The `flexmeasures` integration offers an integration with [FlexMeasures](https://flexmeasures.io/) instances to schedule flexible energy assets.
+
+FlexMeasures can be used to schedule batteries, EV's, heat storage, and other assets. The current integration only support car batteries and batteries optimizations. Shiftable energy assets like dishwashing machines are currently not supported in the integration. The integration offers the following sensor and services:
+
+- A sensor for flexible asset schedules
+- A service to retrieve a schedule
+- A service to post measurements to a FlexMeasures instance
+- A service to change an S2 control type
+
+
+
+To add this integration the following configuration input is needed:
+
+```yaml
+# Example configuration.yaml entry
+flexmeasures:
+  host: "FLEXMEASURES_INSTANCE"
+  username: "USERNAME"
+  password: "PASSWORD"
+  power_sensor: "POWER_SENSOR_ID"
+  consumption_price_sensor: "CONSUMPTION_PRICE_SENSOR_ID"
+  production_price_sensor: "PRODUCTION_PRICE_SENSOR_ID"
+  soc_sensor: "STATE_OF_CHARGE_SENSOR_ID"
+  rm_discharge_sensor: "RESOURCE_MANAGER_DISCHARGE_SENSOR_ID"
+  schedule_duration: "SCHEDULE_DURATION"
+  soc_unit: "STATE_OF_CHARGE_UNIT"
+  soc_min: "STATE_OF_CHARGE_MINIMUM"
+  soc_max: "STATE_OF_CHARGE_MAXIMUM"
+```
+
+{% configuration %}
+host:
+  description: URI of the FlexMeasures instance, for instance (flexmeasures.io)
+  required: true
+  type: string
+username:
+  description: The username associated with the FlexMeasures instance account.
+  required: true
+  type: string
+password:
+  description: The password with the FlexMeasures instance account.
+  required: true
+  type: string
+power_sensor:
+  description: The power sensor that will be scheduled to draw or supply power.
+  required: true
+  type: integer
+consumption_price_sensor:
+  description: The price sensor for the consumption of power to use for the optimization.
+  required: true
+  type: integer
+production_price_sensor:
+  description: The price sensor for the production of power to use for the optimization (this can be the same as the consumption price sensor).
+  required: true
+  type: integer
+soc_sensor:
+  description: The state of charge sensor of the flexible energy asset.
+  required: true
+  type: integer
+rm_discharge_sensor:
+  description: The resource manager discharge sensor.
+  required: true
+  type: integer
+schedule_duration:
+  description: The duration for which the schedules should be calculated in hours.
+  required: true
+  default: 24
```

Comment:
"PT24H"

Reply:
Fixed the type definition of the duration. Good catch